templates/world-map-2d/packages/client/node/src/api.ts

```diff
@@ -0,0 +1,38 @@
+import type { Pool } from "pg";
+import type { StartConfigApiRouter } from "@paimaexample/runtime";
+import type fastify from "fastify";
+import { getUserStats, getAllWorldStats } from "@world-map-2d/db";
+
+export const apiRouter: StartConfigApiRouter = async function (
+  server: fastify.FastifyInstance,
+  dbConn: Pool,
+): Promise<void> {
+  // Get user stats endpoint
+  server.get("/user_stats", async (request, reply) => {
+    const { wallet } = request.query as { wallet: string };
+    if (!wallet) {
+      return reply.code(400).send({ error: "wallet parameter required" });
+    }
+
+    try {
+      const [userStats] = await getUserStats.run({ wallet }, dbConn);
```

Comment:
This need to be wrapped in `runPreparedQuery(...)` 
Our In-memory DB only allows one-thread. 
As we have different process executing DB Queries

https://pglite.dev/docs/
```
As PGlite only has a single exclusive connection to the database, we provide a [multi-tab worker](https://pglite.dev/docs/multi-tab-worker) to enable sharing a PGlite instance between multiple browser tabs.
```

This function runPreparedQuery has a semaphore to allow all systems to share the single connection 💀